include/error_estimation/exact_solution.h

```diff
@@ -342,15 +342,16 @@ class ExactSolution
    * The name of the unknown is
    * the key for the map.
    */
-  typedef std::map<std::string, std::vector<Real>> SystemErrorMap;
+  typedef std::map<std::string, std::vector<Real>, std::less<>>
+    SystemErrorMap;
```

Comment:
Is there a reason to explicitly specify the map comparison function here? Is this somehow not equivalent to`std::less<std::string>` (which would be the default)?

Reply:
It's not, actually!  `std::less<std::string>` is restricted to uses with `std::string`.  `std::less<>` (i.e. [less<void>](https://en.cppreference.com/w/cpp/utility/functional/less_void)) is a new "transparent comparator" functor, which basically compares any two things you throw at it for which `operator<(thing1,thing2)` is well-defined.  If you use the default comparator, then, you can't do operations like `find` with a `string_view`; if you use a transparent comparator you can.

Kind of annoying that using this feature forces you to use extra syntax, but I guess the C++ standards people weren't happy with the fact that making transparent comparators the default could change the behavior of existing code - if you try to do a `find(thing3)`, where thing3 is of some type `sneaky_type &` that's implicitly convertable to `const key_type &`, then with the default comparator it does that implicit conversion and then does `operator<` with the result, whereas with a transparent comparator it just goes straight to `operator<(const key_type &, const sneaky_type &)`.

Reply:
OK, thanks for the explanation. Not particularly self-documenting so you kind of need to be "in the know" to make sense of it. I guess one could explicitly say `std::less<void>` though this is no clearer in my opinion.